src/biconnectivity/articulation.jl

```diff
@@ -69,12 +19,69 @@ julia> articulation(PathGraph(5))
  4
 ```
 """
-function articulation(g::AbstractGraph)
-    state = Articulations(g)
-    for u in vertices(g)
-        if state.depth[u] == 0
-            visit!(state, g, u, u)
+function articulation end
+@traitfn function articulation(g::AG::(!IsDirected)) where {T, AG<:AbstractGraph{T}}
+    s = Vector{Tuple{T, T, T}}()
```

Comment:
Could you use a named tuple here? Otherwise it will be hard to understand this algorithm, especially as it is not exactly the recursive text book one.

Alterative to named tuple, would be to at least add some comments

Reply:
Also `s` is a stack right? So maybe better call that variable `stack`.